diff --git a/source/board/flash/ospi/flash_nor_ospi.c b/source/board/flash/ospi/flash_nor_ospi.c
index 5ad15752c..f87387d90 100644
--- a/source/board/flash/ospi/flash_nor_ospi.c
+++ b/source/board/flash/ospi/flash_nor_ospi.c
@@ -816,7 +816,7 @@ static int32_t Flash_norOspiReadId(Flash_Config *config)
     uint8_t  idCode[FLASH_OSPI_JEDEC_ID_SIZE_MAX];
     uint32_t cmdAddr = OSPI_CMD_INVALID_ADDR;
     uint32_t dummyBits = 0;
-    uint32_t idNumBytes = 3;
+    uint32_t idNumBytes = 3; /* For 1s-1s-1s mode */
     uint32_t numAddrBytes = 0;
 
     if(obj->currentProtocol == FLASH_CFG_PROTO_8D_8D_8D || obj->currentProtocol == FLASH_CFG_PROTO_8S_8S_8S)
@@ -828,7 +828,7 @@ static int32_t Flash_norOspiReadId(Flash_Config *config)
 
         if(obj->currentProtocol == FLASH_CFG_PROTO_8D_8D_8D)
         {
-            idNumBytes = 4; /* Can't read odd bytes in octal DDR */
+            idNumBytes = 6; /* Can't read odd bytes in octal DDR */
         }
     }
     else
@@ -841,21 +841,24 @@ static int32_t Flash_norOspiReadId(Flash_Config *config)
     /* Verify ID with filled data */
     if (status == SystemP_SUCCESS)
     {
-        uint32_t manfID, devID;
-
-        manfID = (uint32_t)idCode[0];
-        devID = ((uint32_t)idCode[1] << 8) | ((uint32_t)idCode[2]);
-        if (!((manfID == config->attrs->manufacturerId) && (devID == config->attrs->deviceId)))
+        uint32_t manfID = 0, devID = 0;
+        if(obj->currentProtocol == FLASH_CFG_PROTO_8D_8D_8D)
         {
-            /* Try the other 3 bytes */
-            manfID = (uint32_t)idCode[3];
-            devID = ((uint32_t)idCode[4] << 8) | ((uint32_t)idCode[5]);
-            if (!((manfID == config->attrs->manufacturerId) && (devID == config->attrs->deviceId)))
+            if((idCode[0] == idCode[1]) && (idCode[2] == idCode[3]) && (idCode[4] == idCode[5]) && (idCode[0] != idCode[2] != idCode[4]))
             {
-                status = SystemP_FAILURE;
+                devID = ((uint32_t)idCode[2] << 8) | ((uint32_t)idCode[4]);
             }
         }
         else
+        {
+            devID = ((uint32_t)idCode[1] << 8) | ((uint32_t)idCode[2]); /* For 1s-1s-1s mode */
+        }
+        manfID = (uint32_t)idCode[0];
+        if (!((manfID == config->attrs->manufacturerId) && (devID == config->attrs->deviceId)))
+        {
+            status = SystemP_FAILURE;
+        }
+        else
         {
             /* Success, nothing to do */;
         }
@@ -1259,7 +1262,14 @@ static int32_t Flash_norOspiOpen(Flash_Config *config, Flash_Params *params)
         OSPI_setCmdExtType(obj->ospiHandle, config->devConfig->cmdExtType);
 
         /* Set initial number of address bytes */
-        obj->numAddrBytes = 3;
+        if(config->devConfig->protocolCfg.protocol == FLASH_CFG_PROTO_8D_8D_8D)
+        {
+            obj->numAddrBytes = 4;
+        }
+        else
+        {
+            obj->numAddrBytes = 3;
+        }
 
         /* Set current protocol as 1s1s1s */
         obj->currentProtocol = FLASH_CFG_PROTO_1S_1S_1S;
